### DISCUSSION TITLE
Forge Scripting Error: 'Could not find target contract

### DISCUSSION THREAD

**Discussion by Bri-3e** [Cyfrin/foundry-full-course-cu#3008] [General]
I'm encountering an error while deploying a contract using Forge:
`forge script script/DeploySimpleStorage.s.sol: DeploySimpleStorage --rpc-url http://localhost:8545 --account defaultkey --sender 0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266 --broadcast -- --vvvv
[⠒] Compiling...
No files changed, compilation skipped
Error: Could not find target contract`

This is my DeploySimpleStorage.s.sol
`// SPDX-License-Identifier: MIT

pragma solidity 0.8.18;

import {Script} from "forge-std/Script.sol";
import {SimpleStorage} from "../src/SimpleStorage.sol";

contract DeploySimpleStorage is Script {
    function run() external returns (SimpleStorage) {
        vm.startBroadcast();
        SimpleStorage simpleStorage = new SimpleStorage();
        vm.stopBroadcast();
        return simpleStorage;
    }
}`
This is my SimpleStorage.sol
`// SPDX-License-Identifier: MIT

pragma solidity ^0.8.18;

// pragma solidity ^0.8.0;
// pragma solidity >=0.8.0 <0.9.0;

contract SimpleStorage {
    uint256 myFavoriteNumber;

    struct Person {
        uint256 favoriteNumber;
        string name;
    }
    // uint256[] public anArray;
    Person[] public listOfPeople;

    mapping(string => uint256) public nameToFavoriteNumber;

    function store(uint256 _favoriteNumber) public virtual {
        myFavoriteNumber = _favoriteNumber;
    }

    function retrieve() public view returns (uint256) {
        return myFavoriteNumber;
    }

    function addPerson(string memory _name, uint256 _favoriteNumber) public {
        listOfPeople.push(Person(_favoriteNumber, _name));
        nameToFavoriteNumber[_name] = _favoriteNumber;
    }
}`

Ive tried checking the contract name and location and recompilling but the error still persist. Im just getting started so ill really appreciate al the help i can get

**Comment by EngrPips** [comment]
Try with this and let me know if it works.

```js
forge script script/DeploySimpleStorage.s.sol:DeploySimpleStorage --rpc-url http://localhost:8545 --account defaultkey --sender 0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266 --broadcast --vvvv
```
The difference between this and yours is that there is a space between the `:` and the `DeploySimpleStorage` in your command and that may cause the issue.

**Reply by LordAdhaar** [reply]
This should work I believe

**Reply by EngrPips** [reply]
Yeah I thought so too.

**Reply by 0xPeregrin** [reply]
Hi @EngrPips. 

I tried cleaning forge and building it again with command foundryup too but Foundry doesn't seem to see my folder after I run
the command below.

Anvil runs in a separate terminal.

forge script script/DeploySimpleStorage.s.sol:DeploySimpleStorage --rpc-url http://localhost:8545

Error: No such file or directory (os error 2)

Thank you!

**Reply by devch136** [reply]
> Hi @EngrPips.
> 
> I tried cleaning forge and building it again with command foundryup too but Foundry doesn't seem to see my folder after I run the command below.
> 
> Anvil runs in a separate terminal.
> 
> forge script script/DeploySimpleStorage.s.sol:DeploySimpleStorage --rpc-url http://localhost:8545
> 
> Error: No such file or directory (os error 2)
> 
> Thank you!

Can you attach your terminal screenshot?

**Reply by EngrPips** [reply]
@0xPeregrin, Please share your working directory screenshot as I share mine below 

![Screenshot 2025-02-28 at 08 54 18](https://github.com/user-attachments/assets/2d5875ba-e674-4e61-bb0d-cded1af7ae77)

**Reply by 0xPeregrin** [reply]
<img width="251" alt="image" src="https://github.com/user-attachments/assets/050cd1bb-c4ba-40dd-96d7-8190ad300a5c" />

@EngrPips there you go.

**Reply by KMean** [reply]
@0xPeregrin make sure to run the command from the correct folder. For example if I run it correctly
```bash
~/projects/cyfrin-updraft-foundry-lottery$ forge script script/DeployRaffle.s.sol:DeployRaffle --rpc-url http://localhost:8545
[⠊] Compiling...
No files changed, compilation skipped
Traces:
  [43671247] → new DeployRaffle@0x9f7cF1d1F558E57ef88a59ac3D47214eF25B6A06
    └─ ← [Return] 217588 bytes of code
```
otherwise
```bash
~/projects$ forge script script/DeployRaffle.s.sol:DeployRaffle --rpc-url http://localhost:8545
Error: No such file or directory (os error 2)
```

**Reply by 0xPeregrin** [reply]
Thank you! @KMean 

my foundry-f23 folder was missing in my code, I added it with cd foundry-f23 before forge command and it worked!

@Heavens01 @EngrPips 

Thanks guys.

**Reply by EngrPips** [reply]
> Thank you! @KMean
> 
> my foundry-f23 folder was missing in my code, I added it with cd foundry-f23 before forge command and it worked!
> 
> @Heavens01 @EngrPips
> 
> Thanks guys.

Awesome, Keep crushing it.

**Comment by Heavens01** [comment]
@Bri-3e 
@0xPeregrin
If you haven't gotten your problem solved yet. Go to your terminal in VSCode and screenshot so I can see the path.

**Reply by EngrPips** [reply]
Great heads-up.
